### PR TITLE
[AV][iOS] Emit fullscreen events for native controls based on view bo…

### PR DESCRIPTION
**Pull request by mczernek**
…unds.

# Why

[Issue #5467](https://github.com/expo/expo/issues/5467)

### CODE REVIEW

File: packages/expo-av/ios/EXAV/Video/EXVideoView.m [hunk 1]
```diff
@@ -241,35 +241,18 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
...
       // Fullscreen player is being dismissed
       _fullscreenPlayerPresented = false;
-      _nativeFullscreenPlayerViewController = nil;
       [self _callFullscreenCallbackForUpdate:EXVideoFullscreenUpdatePlayerWillDismiss];
       [self _callFullscreenCallbackForUpdate:EXVideoFullscreenUpdatePlayerDidDismiss];
-    } else if (fullscreenViewController.isBeingPresented && !_fullscreenPlayerPresented && _nativeFullscreenPlayerViewController == nil) {
+    } else if (viewBounds.size.height == screen.size.height && viewBounds.size.width == screen.size.width && !_fullscreenPlayerPresented) {
       // Fullscreen player is being presented
       _fullscreenPlayerPresented = true;
-      _nativeFullscreenPlayerViewController = fullscreenViewController;
```
Comment (sjchmiela) [inline packages/expo-av/ios/EXAV/Video/EXVideoView.m]: I guess we could remove this variable altogether now.

**Review by sjchmiela — APPROVED** [review]
LGTM!

File: packages/expo-av/ios/EXAV/Video/EXVideoView.m [hunk 2]
```diff
@@ -241,35 +241,18 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
...
+    CGRect screen = [[UIScreen mainScreen] bounds];
+    
+    if (viewBounds.size.height != screen.size.height && viewBounds.size.width != screen.size.width && _fullscreenPlayerPresented) {
       // Fullscreen player is being dismissed
       _fullscreenPlayerPresented = false;
-      _nativeFullscreenPlayerViewController = nil;
       [self _callFullscreenCallbackForUpdate:EXVideoFullscreenUpdatePlayerWillDismiss];
       [self _callFullscreenCallbackForUpdate:EXVideoFullscreenUpdatePlayerDidDismiss];
-    } else if (fullscreenViewController.isBeingPresented && !_fullscreenPlayerPresented && _nativeFullscreenPlayerViewController == nil) {
+    } else if (viewBounds.size.height == screen.size.height && viewBounds.size.width == screen.size.width && !_fullscreenPlayerPresented) {
```
Comment (sjchmiela) [inline packages/expo-av/ios/EXAV/Video/EXVideoView.m]: Oh, what about landscape mode? Then `video.height == screen.width` etc…

Comment (sjchmiela) [reply packages/expo-av/ios/EXAV/Video/EXVideoView.m]: Or does `UIScreen.mainScreen` handle this?

Comment (mczernek) [reply packages/expo-av/ios/EXAV/Video/EXVideoView.m]: I've tested it in landscape, both with locked and unlocked orientation and it works just fine.

File: packages/expo-av/ios/EXAV/Video/EXVideoView.m (hunk 1 above)
Comment (mczernek) [reply packages/expo-av/ios/EXAV/Video/EXVideoView.m]: Yes we can!

**Review by bbarthec — APPROVED** [review]
CHANGELOG 🙏 😉

**Pull request merged**